pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -64,94 +93,134 @@ func NewPortTable(start, end int) (*PortTable, error) {
 func (pt *PortTable) CleanupAllEntries() {
 	pt.tableLock.Lock()
 	defer pt.tableLock.Unlock()
-	pt.Table = make(map[int]NodePortData)
+	pt.Table = make(map[int]*NodePortData)
+	pt.PodTable = make(map[string]*NodePortData)
 }
 
 func (pt *PortTable) GetEntry(nodeport int) *NodePortData {
```

Comment:
this function is not called? Could we remove it given that it's no longer thread-safe now that we return a pointer to `NodePortData` instead of a pointer to a copy?